### PR TITLE
[mono][ios] Revert LLVM mode for apple mobile tests

### PR DESCRIPTION
**Pull request by kotlarmilos**
This PR reverts using LLVM mode for apple mobile tests. The regression was introduced in https://github.com/dotnet/runtime/pull/87773/files#diff-efa78f3f0523ef8b61f94772d9c204608051d411527cb9be83ca60266dc1d2f9R19.

### CODE REVIEW

**Comment by kotlarmilos** [comment]
We probably need to backport it to .NET 8.

**Review by steveisok — APPROVED** [review]

**Review by ivanpovazan — APPROVED** [review]

File: eng/testing/tests.ioslike.targets [hunk 1]
```diff
@@ -16,7 +16,7 @@
     <!-- running aot-helix tests locally, so we can test with the same project file as CI -->
     <_AOTBuildCommand Condition="'$(ContinuousIntegrationBuild)' != 'true'">$(_AOTBuildCommand) /p:RuntimeSrcDir=$(RepoRoot) /p:RuntimeConfig=$(Configuration)</_AOTBuildCommand>
     
-    <_AOTBuildCommand>$(_AOTBuildCommand) /p:XHARNESS_EXECUTION_DIR=&quot;$XHARNESS_EXECUTION_DIR&quot; /p:RunAOTCompilation=$(RunAOTCompilation) /p:UseNativeAOTRuntime=$(UseNativeAOTRuntime) /p:TargetOS=$(TargetOS) /p:TargetArchitecture=$(TargetArchitecture) /p:MonoForceInterpreter=$(MonoForceInterpreter) /p:DevTeamProvisioning=$(DevTeamProvisioning) /p:UsePortableRuntimePack=true /p:Configuration=$(Configuration)</_AOTBuildCommand>
+    <_AOTBuildCommand>$(_AOTBuildCommand) /p:XHARNESS_EXECUTION_DIR=&quot;$XHARNESS_EXECUTION_DIR&quot; /p:RunAOTCompilation=$(RunAOTCompilation) /p:UseNativeAOTRuntime=$(UseNativeAOTRuntime) /p:TargetOS=$(TargetOS) /p:TargetArchitecture=$(TargetArchitecture) /p:MonoForceInterpreter=$(MonoForceInterpreter) /p:MonoEnableLLVM=true /p:DevTeamProvisioning=$(DevTeamProvisioning) /p:UsePortableRuntimePack=true /p:Configuration=$(Configuration)</_AOTBuildCommand>
```
Comment (ivanpovazan) [inline eng/testing/tests.ioslike.targets:19]: Might be good to add a comment what is the reason of passing `/p:MonoEnableLLVM=true` here, so we don't end up trying to remove it is some later refactoring

**Comment by SamMonoRT** [comment]
> We probably need to backport it to .NET 8.

Yes, backport to RC1 following standard backport proccess

**Review by SamMonoRT — APPROVED** [review]

**Pull request merged**